catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -2482,73 +2418,43 @@ protected ResourceResponse getResource(ResourceRequest resourceRequest, boolean
+                LOGGER.debug("metacard for product found on source: {}", resolvedSourceId);
+                FederatedSource source;
 
-            if (resourceResponse == null) {
-                // retrieve product from specified federated site if not in cache
-                if (!resourceSourceName.equals(getId())) {
-                    LOGGER.debug("Searching federatedSource {} for resource.", resourceSourceName);
-                    LOGGER.debug("metacard for product found on source: {}", resolvedSourceId);
-                    FederatedSource source;
+                source = frameworkProperties.getFederatedSources()
+                        .get(resourceSourceName);
 
-                    source = frameworkProperties.getFederatedSources()
-                            .get(resourceSourceName);
-                    if (source != null) {
-                        LOGGER.debug("Adding federated site to federated query: {}",
-                                source.getId());
-                    }
-
-                    if (source != null) {
-                        LOGGER.debug("Retrieving product from remote source {}", source.getId());
-                        ResourceRetriever retriever = new RemoteResourceRetriever(source,
-                                responseURI,
-                                requestProperties);
-                        try {
-                            resourceResponse =
-                                    frameworkProperties.getReliableResourceDownloadManager()
-                                            .download(resourceRequest, metacard, retriever);
-                        } catch (DownloadException e) {
-                            LOGGER.info("Unable to download resource", e);
-                        }
-                    } else {
-                        LOGGER.warn("Could not find federatedSource: {}", resourceSourceName);
-                    }
-                } else {
-                    LOGGER.debug("Retrieving product from local source {}", resourceSourceName);
-                    ResourceRetriever retriever =
-                            new LocalResourceRetriever(frameworkProperties.getResourceReaders(),
-                                    responseURI,
-                                    requestProperties);
+                if (source != null) {
+                    LOGGER.debug("Adding federated site to federated query: {}",
+                            source.getId());
+                    LOGGER.debug("Retrieving product from remote source {}", source.getId());
+                    ResourceRetriever retriever = new RemoteResourceRetriever(source,
+                            responseURI,
+                            requestProperties);
                     try {
-                        resourceResponse = frameworkProperties.getReliableResourceDownloadManager()
-                                .download(resourceRequest, metacard, retriever);
+                        resourceResponse =
```

Comment:
This try block looks like the same code in the else case.  Can this be pulled outside of this if-else block?